modules/nextflow/src/main/groovy/nextflow/k8s/model/PodOptions.groovy

```diff
@@ -109,6 +111,9 @@ class PodOptions {
         else if( entry.nodeSelector ) {
             this.nodeSelector = new PodNodeSelector(entry.nodeSelector)
         }
+        else if( entry.affinity ) {
```

Comment:
It may be better to check `if ( entry.affinity instanceof Map )`, to prevent an invalid type is forced into a map which can be a bit undetermined the result